lib/templatesimplifier.cpp

```diff
@@ -2044,7 +2044,7 @@ void TemplateSimplifier::expandTemplate(
                     if (isVariadicTemplateArg && Token::Match(tok3, "%name% ... %name%"))
                         tok3 = tok3->tokAt(2);
                     const std::string endStr(isVariadicTemplateArg ? ">" : ",>");
-                    for (const Token *typetok = mTypesUsedInTemplateInstantiation[itype].token();
+                    for (Token *typetok = mTypesUsedInTemplateInstantiation[itype].token();
```

Comment:
hmm.. if typetok is not const then there is no reason to const_cast it.

Reply:
Done.

I did not check the surrounding code yet. I wanted to do that after the methods are proper to avoid potential flip-flopping on the specifiers.